Oracle/PCORNetInit.sql

```diff
@@ -0,0 +1,1000 @@
+  ELSE
+   --dbms_output.put_line('YES!'); 
+   return TRUE;
+  END IF;
+
+END PMN_IFEXISTS;
+/
+
+
+create or replace PROCEDURE PMN_Execuatesql(sqlstring VARCHAR2) AS 
+BEGIN
+  EXECUTE IMMEDIATE sqlstring;
+  dbms_output.put_line(sqlstring);
+END PMN_ExecuateSQL;
+/
+
+
+--ACK: http://dba.stackexchange.com/questions/9441/how-to-catch-and-handle-only-specific-oracle-exceptions
+create or replace procedure create_error_table(table_name varchar2) as
+sqltext varchar2(4000); 
+
+begin
+  dbms_errlog.create_error_log(dml_table_name => table_name);
+EXCEPTION
+  WHEN OTHERS THEN
+    IF SQLCODE = -955 THEN
+      NULL; -- suppresses ORA-00955 exception ("name is already used by an existing object")
+    ELSE
+       RAISE;
+    END IF;
+-- Delete rows from a previous run in case the table already existed
+sqltext := 'delete from ERR$_' || table_name;
+PMN_Execuatesql(sqltext);
+end;
+/
+
+
+create or replace FUNCTION GETDATAMARTID RETURN VARCHAR2 IS 
+BEGIN 
+    RETURN '&&datamart_id';
+END;
+/
+
+
+CREATE OR REPLACE FUNCTION GETDATAMARTNAME RETURN VARCHAR2 AS 
+BEGIN 
+    RETURN '&&datamart_name';
+END;
+/
+
+
+CREATE OR REPLACE FUNCTION GETDATAMARTPLATFORM RETURN VARCHAR2 AS 
+BEGIN 
+    RETURN '02'; -- 01 is MSSQL, 02 is Oracle
+END;
+/
+
+
+--------------------------------------------------------------------------------
+-- I2B2 SYNONYMS, VIEWS, AND INTERMEDIARY TABLES
+--------------------------------------------------------------------------------
+
+
+CREATE OR REPLACE SYNONYM I2B2FACT FOR "&&i2b2_data_schema".OBSERVATION_FACT
+/
+
+CREATE OR REPLACE SYNONYM I2B2MEDFACT FOR OBSERVATION_FACT_MEDS
+/
+
+BEGIN
+PMN_DROPSQL('DROP TABLE i2b2patient_list');
+END;
+/
+
+CREATE table i2b2patient_list as 
+select * from
+(
+select DISTINCT PATIENT_NUM from I2B2FACT where START_DATE > to_date('&&min_pat_list_date_dd_mon_rrrr','dd-mon-rrrr')
+) where ROWNUM<100000000
+/
+
+create or replace VIEW i2b2patient as select * from "&&i2b2_data_schema".PATIENT_DIMENSION where PATIENT_NUM in (select PATIENT_NUM from i2b2patient_list)
+/
+
+create or replace view i2b2visit as select * from "&&i2b2_data_schema".VISIT_DIMENSION where START_DATE >= to_date('&&min_visit_date_dd_mon_rrrr','dd-mon-rrrr') and (END_DATE is NULL or END_DATE < CURRENT_DATE) and (START_DATE <CURRENT_DATE)
+/
+
+CREATE OR REPLACE SYNONYM pcornet_med FOR  "&&i2b2_meta_schema".pcornet_med
+/
+
+CREATE OR REPLACE SYNONYM pcornet_lab FOR  "&&i2b2_meta_schema".pcornet_lab
+/
+
+CREATE OR REPLACE SYNONYM pcornet_diag FOR  "&&i2b2_meta_schema".pcornet_diag
+/
+
+CREATE OR REPLACE SYNONYM pcornet_demo FOR  "&&i2b2_meta_schema".pcornet_demo
+/
+
+CREATE OR REPLACE SYNONYM pcornet_proc FOR  "&&i2b2_meta_schema".pcornet_proc
+/
+
+CREATE OR REPLACE SYNONYM pcornet_vital FOR  "&&i2b2_meta_schema".pcornet_vital
+/
+
+CREATE OR REPLACE SYNONYM pcornet_enc FOR  "&&i2b2_meta_schema".pcornet_enc
+/
+
+
+--------------------------------------------------------------------------------
+-- I2P REPORT
```

Comment:
I think I saw something in the SCILHS issue tracker that I2P REPORT is now dead code. I'd prefer not to maintain it. Move it to its own file?